network/common/src/main/java/org/apache/spark/network/sasl/SaslClientBootstrap.java

```diff
@@ -106,4 +115,29 @@ public void doBootstrap(TransportClient client, Channel channel) {
     }
   }
 
+  /**
+   * Negotiates AES cipher based on complete {@link SparkSaslClient}. The keys need to be
+   * decrypted by sasl client.
+   */
+  private void negotiateAes(TransportClient client, SparkSaslClient saslClient) throws IOException {
+    // create option for negotiation
+    CipherOption cipherOption = new CipherOption(conf.saslEncryptionAesCipherTransformation());
+    ByteBuf buf = Unpooled.buffer(cipherOption.encodedLength());
+    cipherOption.encode(buf);
+
+    // send option to server and decode received negotiated option
+    ByteBuffer response = client.sendRpcSync(buf.nioBuffer(), conf.saslRTTimeoutMs());
+    cipherOption = CipherOption.decode(Unpooled.wrappedBuffer(response));
+
+    // decrypt key from option. Server's outKey is client's inKey, and vice versa.
+    byte[] outKey = saslClient.unwrap(cipherOption.inKey, 0, cipherOption.inKey.length);
+    byte[] inKey = saslClient.unwrap(cipherOption.outKey, 0, cipherOption.outKey.length);
+
+    // enable AES on saslClient
+    Properties properties = new Properties();
```

Comment:
hints: Let's make it simple by combing L137-L139 since we don't add any property actually.
```
    saslClient.enableAes(CipherTransformation.fromName(cipherOption.cipherSuite), new Properties(), 
       inKey, outKey, cipherOption.outIv, cipherOption.inIv);
```